src/DataStructures/Tensor/Tensor.hpp

```diff
@@ -494,3 +495,93 @@ std::ostream& operator<<(std::ostream& os,
      << "\n     Multiplicity: " << x.multiplicity(i) << " Index: " << i;
   return os;
 }
+
+namespace MakeWithValueImpl_detail {
+/// \ingroup DataStructures
+/// \brief Returns a Tensor whose DataVectors are the same size as `input`, with
+/// each element equal to `value`.
+template <typename Symm, template <typename...> class IndexList,
```

Comment:
Just do `Tensor<DataVector, Args...>`, then we don't need the template-template parameter. You don't actually care about the indices themselves so there's no point in make the compiler do the extra work. This applies to all similar below.